src/OpenLoco/src/Audio/Audio.cpp

```diff
@@ -1027,68 +1027,94 @@ namespace OpenLoco::Audio
         }
     }
 
-    static void addAllSongsToPlaylist(int32_t excludeTrack, std::vector<uint8_t>& playlist)
+    static void addAllSongsToPlaylist(std::vector<uint8_t>& playlist)
     {
         for (auto i = 0; i < kNumMusicTracks; i++)
         {
-            if (i != excludeTrack)
-            {
-                playlist.push_back(i);
-            }
+            playlist.push_back(i);
         }
     }
 
-    static void addCurrentEraSongsToPlaylist(int32_t excludeTrack, std::vector<uint8_t>& playlist)
+    static void addCurrentEraSongsToPlaylist(std::vector<uint8_t>& playlist)
     {
         auto currentYear = getCurrentYear();
         for (auto i = 0; i < kNumMusicTracks; i++)
         {
             const auto& mi = kMusicInfo[i];
             if (currentYear >= mi.startYear && currentYear <= mi.endYear)
             {
-                if (i != excludeTrack)
-                {
-                    playlist.push_back(i);
-                }
+                playlist.push_back(i);
             }
         }
     }
 
-    static int32_t chooseNextMusicTrack(int32_t excludeTrack)
+    static void addCustomSelectionSongsToPlaylist(std::vector<uint8_t>& playlist)
     {
-        using MusicPlaylistType = Config::MusicPlaylistType;
-
-        static std::vector<uint8_t> playlist;
-        playlist.clear();
-
         const auto& cfg = Config::get().old;
-        switch (cfg.musicPlaylist)
+        for (auto i = 0; i < kNumMusicTracks; i++)
+        {
+            if (cfg.enabledMusic[i] & 1)
+            {
+                playlist.push_back(i);
+            }
+        }
+    }
+
+    static void addSelectedPlaylistToPlaylist(std::vector<uint8_t>& playlist)
```

Comment:
Every single user of this function is passing in an empty vector so why even bother appending. It would be a better design to just return a new vector then you can get rid of make selected playlist.

Reply:
That sounds like `makeSelectedPlaylist`, so I'm interpreting this as deleting `addSelectedPlaylistToPlaylist` and in-lining its code into `makeSelectedPlaylist`